tensorflow/core/kernels/mkl/mkl_quantize_op.cc

```diff
@@ -457,7 +457,7 @@ class MklQuantizeV2Op : public OpKernel {
         // If it is signed, we try to keep 0.0 being 0 and drop one bucket. For
         // example, if it is 8 bits, we have the range [-127, 127]. So for input
         // range of [-x, x], the scale should be 254/(2*x).
-        target_range = static_cast<float>((uint64_t{1} << (num_bits - 1)) - 1);
+        target_range = static_cast<float>((uint64_t{1} << num_bits) - 1) / 2.;
```

Comment:
Why is this change needed? tests seem to pass without this change. Also, this change caused some results mismatch in //tensorflow/core/kernels/mkl:mkl_quantize_op_test (this test only runs with --config=mkl).

Reply:
Without this change the test //tensorflow/python:quantized_ops_test fails. The following code is part of the test: 
```
#!/usr/bin/python3

import numpy as np
from tensorflow.python.framework import constant_op
from tensorflow.python.ops import array_ops
from tensorflow.python.framework import dtypes

values = np.array([-1, -0.5, 0.555, 1], dtype=np.float32)
quant_values = np.array([-128, -64, 71, 127], dtype=np.int32)

inputs = constant_op.constant(values)
expected_quantized = quant_values

min_range, max_range = -1.0, 0.8

quantized =  array_ops.quantize(
    inputs,
    min_range,
    max_range,
    T=dtypes.qint8,
    mode="SCALED",
    round_mode="HALF_TO_EVEN",
    axis=None).output

print('Returned: ', quantized.numpy())
print('Expected: ', expected_quantized)
```
Without above change the returned values are `[-127, -64, 70, 127]` instead of `[-128, -64, 71, 127]`. Also the above formula is formula that is here: https://intellabs.github.io/distiller/algo_quantization.html for calculating scale factor for full range.